lib/mayaUsd/ufe/UsdContextOps.cpp

```diff
@@ -54,6 +56,58 @@ class SetVariantSelectionUndoableCommand : public Ufe::UndoableCommand
     const std::string fNewSelection;
 };
 
+//! \brief Undoable command for add new prim
+class AddNewPrimUndoableCommand : public Ufe::UndoableCommand
+{
+public:
+    AddNewPrimUndoableCommand(const MayaUsd::ufe::UsdSceneItem::Ptr& usdSceneItem,
+                              const Ufe::ContextOps::ItemPath& itemPath)
+    : Ufe::UndoableCommand()
+    {
+        // First get the stage from the proxy shape.
+        auto ufePath = usdSceneItem->path();
+        auto segments = ufePath.getSegments();
+        auto dagSegment = segments[0];
+        _stage = MayaUsd::ufe::getStage(Ufe::Path(dagSegment));
+        if (_stage) {
+            // Rename the new prim for uniqueness, if needed.
+            Ufe::Path newUfePath = ufePath + itemPath[1];
+            auto newPrimName = uniqueChildName(usdSceneItem, newUfePath);
+
+            // Build (and store) the path for the new prim with the unique name.
+            PXR_NS::SdfPath usdItemPath = usdSceneItem->prim().GetPath();
+            _primPath = usdItemPath.AppendChild(PXR_NS::TfToken(newPrimName));
+
+            // The type of prim we were asked to create.
+            if (itemPath[1] == "Def")
+                _primToken = TfToken();     // create typeless prim
```

Comment:
Nit-pick: could be in the constructor's initializer list as

_primToken(itemPath[1] == "Def" ? TfToken() : TfToken(itemPath[1]))

Reply:
O.K.